astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
+    PyUnitListProxy *b,
+    int op) {
+
+  int equal;
+  int status;
```

Comment:
As said before I would prefer not to have `status` and fail immediately. That makes it much easier to reason about the code.